test/xpu_api/utilities/optional/optional.object/optional.object.ctor/explicit_optional_U.pass.cpp

```diff
@@ -0,0 +1,90 @@
+// -*- C++ -*-
+//===----------------------------------------------------------------------===//
+//
+// Copyright (C) Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// This file incorporates work covered by the following copyright and permission
+// notice:
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+//
+//===----------------------------------------------------------------------===//
+
+#include "support/test_config.h"
+
+#include <oneapi/dpl/optional>
+#include <oneapi/dpl/utility>
+#include <oneapi/dpl/type_traits>
+
+#include "support/test_macros.h"
+#include "support/utils.h"
+
+#if TEST_DPCPP_BACKEND_PRESENT
+using dpl::optional;
+
+template <class KernelTest, class T, class U>
+bool
+test(optional<U>&& rhs, bool is_going_to_throw = false)
```

Comment:
Consider creating optional object inside of the function instead of passing as rvalue reference.

Reply:
It's look like not relevant change.

Reply:
Fixed